src/main/java/gigaherz/toolbelt/customslots/ExtensionSlotItemHandler.java

```diff
@@ -42,7 +42,8 @@ public ResourceLocation getType()
     @Override
     public ItemStack getContents()
     {
-        return inventory.getStackInSlot(slot);
+        ItemStack result = inventory.getStackInSlot(slot);
+        return result == null ? ItemStack.EMPTY : result;
```

Comment:
Err, this is my fault I guess. I should have been more clear: the situation where *a slot contains null* should never happen, so this still shouldn't be needed. :P
We need to track down which part of the code is setting the slot to null, because that part of the code is at fault.

Reply:
Oh okay!

Reply:
I should note that it's happening on a fresh life, i.e. I suspect that the slot is being initialized to null.

Reply:
Well, I'm using forge's ItemStackHandler, which uses a NonNullList set to get initialized to ItemStack.EMPTY.... so that's unlikely.

I'm gonna add quark locally and try it myself, see if I can catch who's setting things to null.

Reply:
Yeah, the ItemStack contract is that null will never be returned, but from what I understood of the code associated with that Quark bug (which was actually fixed in a patch) and [this](https://github.com/Vazkii/Botania/issues/2939) issue with Botania (and an adjacent issue with Electroblob's Wizardry), it's [this](https://github.com/Vazkii/Botania/blob/master/src/main/java/vazkii/botania/common/item/ItemKeepIvy.java#L44) line that's problematic -- a null being inserted into the EntityItem list. I haven't had a chance to actually look through gchpaco's PR to see what's being addressed to see how this directly relates to the EntityItems.

Reply:
PROBLEM FOUND! I forgot to add a check to make sure the slot is not empty.
dropItem should only ever be called if the slot contains a non-empty stack!

Thanks for the attempt @gchpaco, I can take it from here, it's just literally two lines that I already fixed locally while trying things. :P